packages/core/src/lib/modal.svelte

```diff
@@ -0,0 +1,108 @@
+<!--
+@component
+
+Creates a modal overlay.
+
+```svelte
+    <Modal
+      open={modalOpen}
+      title='This is the modal demo'
+      message='Are you sure you want to kick off a notify toast?'
+      on:close={handleCloseModal}
+    >
+      <Button slot='primary' on:click={() => notify.success('Howdy Message', 'Howdy Partner')}>Notify howdy</Button>
+      <Button
+        slot='secondary'
+        variant='dark'
+        on:click={handleCloseModal}
+      >Cancel</Button>
+    </Modal>
+```
+-->
+<svelte:options immutable />
+
+<script lang="ts">
+import cx from 'classnames';
+import { createEventDispatcher, onMount } from 'svelte';
+import IconButton from './button/icon-button.svelte';
+
+/** The title text in the modal. */
+export let title = '';
+/** The message text in the modal. */
+export let message = '';
+/** Whether the modal is open. */
+export let open = false;
+/** The variant of the modal. */
+export let variant: 'small' | '' = '';
+
+const dispatch = createEventDispatcher<{
+  /** Fires when the modal is requested to close. */
+  close: boolean;
+}>();
+
+const handleCloseModal = () => {
+  dispatch('close', true);
+};
+
+const handleBackgroundClick = (event: MouseEvent) => {
+  if (event.target === event.currentTarget) {
+    handleCloseModal();
+  }
+};
+
+const handleBackgroundKey = (event: KeyboardEvent) => {
+  if (event.key === 'Enter' || event.key === 'Space') {
+    handleCloseModal();
+  }
+};
+
+onMount(() => {
+  document.body.classList.toggle('overflow-hidden', open);
+});
+</script>
+
+<div
```

Comment:
I'm getting some A11y errors.  For example, A11y: Non-interactive element <div> should not be assigned mouse or keyboard event listeners.(a11y-no-noninteractive-element-interactions).  Should I make this div a button?  If I do that, the cursor makes the background look clickable, which looks a bit strange.  Anyone have any suggestions?

Reply:
For the mouse handler, you could use [`clickOutside`](https://github.com/viamrobotics/prime/blob/main/packages/core/src/lib/click-outside.ts) on the modal element.

For the keyboard handler, you could add the handler `onMount` to the `window`.

Reply:
Ty ty!  Will try those!